d3/js/AliTV.js

```diff
@@ -263,35 +271,41 @@ AliTV.prototype.getLinearLinkCoords = function(coords) {
 		var karyo2Coords = coords[karyoMap[feature2.karyo]];
 		var genomePosition1 = that.filters.karyo.genome_order.indexOf(karyo1.genome_id);
 		var genomePosition2 = that.filters.karyo.genome_order.indexOf(karyo2.genome_id);
-
-		if (genomePosition1 > genomePosition2) {
-			var tmp = feature1;
-			feature1 = feature2;
-			feature2 = tmp;
-			tmp = karyo1;
-			karyo1 = karyo2;
-			karyo2 = tmp;
-			tmp = karyo1Coords;
-			karyo1Coords = karyo2Coords;
-			karyo2Coords = tmp;
-		}
-		link.source0.x = karyo1Coords.x + karyo1Coords.width * feature1.start / karyo1.length;
-		link.source0.y = karyo1Coords.y + karyo1Coords.height + conf.graphicalParameters.linkKaryoDistance;
-		link.source1.x = karyo1Coords.x + karyo1Coords.width * feature1.end / karyo1.length;
-		link.source1.y = karyo1Coords.y + karyo1Coords.height + conf.graphicalParameters.linkKaryoDistance;
-
-		link.target0.x = karyo2Coords.x + karyo2Coords.width * feature2.start / karyo2.length;
-		link.target0.y = karyo2Coords.y - conf.graphicalParameters.linkKaryoDistance;
-		link.target1.x = karyo2Coords.x + karyo2Coords.width * feature2.end / karyo2.length;
-		link.target1.y = karyo2Coords.y - conf.graphicalParameters.linkKaryoDistance;
-
-		if (Math.abs(genomePosition2 - genomePosition1) === 1) {
-			link.adjacent = true;
-			linearLinkCoords.push(link);
-		} else {
-			link.adjacent = false;
-			if (conf.linear.drawAllLinks === true) {
-				linearLinkCoords.push(link);
+		var lengthOfFeature1 = that.data.features[value.source].end - that.data.features[value.source].start;
```

Comment:
```lengthOfFeature1``` might end up being negative (if it is on the reverse strand, start>end).
In this case it will be outside of the allowed range despite its absolute value being okay.
Use ```Math.abs()``` to make negative numbers positive.